process/block/shardblock.go

```diff
@@ -1844,3 +1891,57 @@ func (sp *shardProcessor) getBootstrapHeadersInfo(
 
 	return lastSelfNotarizedHeaders
 }
+
+func (sp *shardProcessor) receivedMiniBlock(key []byte) {
+	log.Debug("miniblock key", "key", key)
+	mb, ok := sp.dataPool.MiniBlocks().Get(key)
+	if ok {
+		peerMb, ok := mb.(*block.MiniBlock)
+		if ok && peerMb.Type == block.PeerBlock {
+			log.Debug(fmt.Sprintf("received miniblock of type %s", peerMb.Type))
+
+			sp.startOfEpochDataMutex.Lock()
+
+			defer sp.startOfEpochDataMutex.Unlock()
+			for _, sod := range sp.startOfEpochData {
+				if sod.stillMissing > 0 {
+					for _, mbHash := range sod.peerMiniblocks {
+						if bytes.Equal(mbHash, key) {
+							sod.stillMissing--
+						}
+					}
+				}
+
+				if sod.stillMissing == 0 {
+					if sp.processPeerMiniBlocks(sod) {
+						return
+					}
+
+					sp.receivedMetaBlock(sod.metaHeader, sod.headerHash)
+					return
+				}
+			}
+		}
+	}
+}
+
+func (sp *shardProcessor) processPeerMiniBlocks(sod startOfEpochData) bool {
```

Comment:
move all the peer processing to the validator statistics pre processor.

Reply:
fixed